### PR TITLE
Allow HMR with virtual sites in OpenMM

### LINKED ISSUE #1055: [DOCS] Document HMR virtual site limitation
**Description**

Virtual sites are not currently allowed when doing HMR, however there is no documentation at the "to_openmm_system" docstring level (or elsewhere as far as I can tell). This means you have to encounter the error to know the limitation.

https://github.com/openforcefield/openff-interchange/blob/57dd53aca23c0a038b09c6d2a26911388a4e64d8/openff/interchange/interop/openmm/__init__.py#L183-L187

**The ask**

A little bit more docstring docs here: https://github.com/openforcefield/openff-interchange/blob/57dd53aca23c0a038b09c6d2a26911388a4e64d8/openff/interchange/interop/openmm/__init__.py#L56-L57

### LINKED ISSUE #1056: [ENH] Add support for HMR with virtual sites
**Description**

I notice the wording is "not yet": https://github.com/openforcefield/openff-interchange/blob/57dd53aca23c0a038b09c6d2a26911388a4e64d8/openff/interchange/interop/openmm/__init__.py#L186

I also don't seem to see any open issues, so erm.. here's an issue for it 🙃

### PR DESCRIPTION
**Pull request by mattwthompson**
### Description

Resolves #1056

Resolves #1055

### Checklist

- [x] Add tests
- [x] Lint
- [x] Update docstrings

### CODE REVIEW

File: openff/interchange/_tests/unit_tests/interop/openmm/test_hmr.py [hunk 1]
```diff
@@ -80,6 +81,57 @@ def test_mass_is_positive(sage):
...
+    assert system.getParticleMass(4)._value == hydrogen_mass
+    assert system.getParticleMass(5)._value == hydrogen_mass
+    assert system.getParticleMass(6)._value == hydrogen_mass
+    assert system.getParticleMass(7)._value == hydrogen_mass
+
+    # make sure it can minimize and run briefly without crashing
+    # this is a single ligand in vacuum with boring parameters
+    simulation.minimizeEnergy(maxIterations=100)
+
+    simulation.runForClockTime(10 * openmm.unit.second)
```
Comment (Yoshanuikabundi) [inline openff/interchange/_tests/unit_tests/interop/openmm/test_hmr.py:137]: Do these tests need equivalents for the GROMACS exporter?

File: openff/interchange/_tests/unit_tests/interop/openmm/test_hmr.py [hunk 2]
```diff
@@ -80,6 +81,57 @@ def test_mass_is_positive(sage):
...
+    assert system.getParticleMass(0)._value == 35.4532
+    assert system.getParticleMass(3)._value == 35.4532
+
+    assert system.getParticleMass(1)._value == 12.01078 - 2 * (hydrogen_mass - 1.007947)
+    assert system.getParticleMass(2)._value == 12.01078 - 2 * (hydrogen_mass - 1.007947)
+
+    assert system.getParticleMass(4)._value == hydrogen_mass
+    assert system.getParticleMass(5)._value == hydrogen_mass
+    assert system.getParticleMass(6)._value == hydrogen_mass
+    assert system.getParticleMass(7)._value == hydrogen_mass
```
Comment (Yoshanuikabundi) [inline openff/interchange/_tests/unit_tests/interop/openmm/test_hmr.py:131]: In the issue #1056 you note that "tests should make sure... Virtual sites continue to never have any mass". Should that be checked here?

**Review by Yoshanuikabundi — COMMENTED** [review]
My test of this behaviour worked as expected (and looked a lot like your test), so I think this is in good shape! Definitely a great feature.

My notes are mostly a repeated typo in existing code, improved linking, and nits around tests. The one thing I would really like to see addressed is some confusion around caveats in the status page.

I have one API ergonomics notes that may be out of scope here: I expected to be able to provide units to the `hydrogen_mass` argument, but this raised two different and cryptic errors depending on if I used openmm or openff units. It would be great if mass quantities with units could be accepted in addition to raw floats.

File: openff/interchange/_tests/unit_tests/interop/openmm/test_hmr.py (hunk 1 above)
Comment (mattwthompson) [reply openff/interchange/_tests/unit_tests/interop/openmm/test_hmr.py:137]: No ... the GROMACS code didn't actually change, so I'm not sure why these docstrings did.

**Pull request merged**